From: Gordon Malm <gengor@gentoo.org>

Strip keyboard hot-plugging support out of systemsettings' kxkb module.
Systemsettings' keyboard hot-plugging code relies on override abilities
found only in GCC 4.1.2 and later.

See: http://bugs.gentoo.org/256827

--- a/kcontrol/kxkb/CMakeLists.txt
+++ b/kcontrol/kxkb/CMakeLists.txt
@@ -51,9 +51,6 @@ if(USE_XKLAVIER STREQUAL ON)
 endif(USE_XKLAVIER STREQUAL ON)
 macro_log_feature(X11_Xkbfile_FOUND "xkbfile" "X11 KXB library for keyboard handling software" "http://www.x.org" FALSE "" "xkbfile is required to build kxkb.")
 
-macro_bool_to_01(X11_Xinput_FOUND HAVE_XINPUT)
-ADD_DEFINITIONS(-DHAVE_XINPUT=${HAVE_XINPUT})
-
 if(X11_Xkbfile_FOUND AND HAVE_XKLAVIER)
 
 #MESSAGE("xkb support libs:: ${XKB_SUPPORT_LIB}")
--- a/kcontrol/kxkb/x11helper.cpp
+++ b/kcontrol/kxkb/x11helper.cpp
@@ -425,7 +425,7 @@ X11Helper::isNewDeviceEvent(XEvent* even
 int
 X11Helper::registerForNewDeviceEvent(Display* display)
 {
-    kWarn() << "Kxkb is compiled without XInput, xkb configuration will be reset when new keyboard device is plugged in!";
+    kWarning() << "Kxkb is compiled without XInput, xkb configuration will be reset when new keyboard device is plugged in!";
     return -1;
 }
 #endif
--- a/kcontrol/kxkb/x11helper.h
+++ b/kcontrol/kxkb/x11helper.h
@@ -64,9 +64,9 @@ public:
 
 	static QString getWindowClass(Window winId, Display* dpy);
 
-#ifdef HAVE_XINPUT
         static int registerForNewDeviceEvent(Display* dpy);
         static int isNewDeviceEvent(XEvent* event);
+#ifdef HAVE_XINPUT
         static int m_xinputEventType;
 #endif
 
